spec/rails_helper.rb

```diff
@@ -82,7 +92,48 @@
 Capybara.default_driver = :rack_test # This is a faster driver
 Capybara.javascript_driver = :selenium_chrome_headless_sandboxless # This is slower
 
+Capybara::Screenshot.register_driver(:selenium_chrome_headless_sandboxless) do |driver, path|
+  driver.browser.save_screenshot(path)
+end
+
+Capybara::Screenshot.autosave_on_failure = false
+
+# Save CircleCI artifacts
+
+def save_timestamped_page_and_screenshot(page, meta)
+  filename = File.basename(meta[:file_path])
+  line_number = meta[:line_number]
+
+  time_now = Time.now
+  timestamp = "#{time_now.strftime('%Y-%m-%d-%H-%M-%S.')}#{'%03d' % (time_now.usec/1000).to_i}"
+
+  screenshot_name = "screenshot-#{filename}-#{line_number}-#{timestamp}.png"
+  screenshot_path = "#{Rails.root.join('tmp/capybara')}/#{screenshot_name}"
+  page.save_screenshot(screenshot_path)
+
+  page_name = "html-#{filename}-#{line_number}-#{timestamp}.html"
+  page_path = "#{Rails.root.join('tmp/capybara')}/#{page_name}"
+  page.save_page(page_path)
+
+  puts "\n  Screenshot: tmp/capybara/#{screenshot_name}"
+  puts "  HTML: tmp/capybara/#{page_name}"
+end
+
```

Comment:
end stuff that belongs in a support file

Reply:
Should I not just put all of the capybara set up into its own file?

Reply:
yes, that is what I was suggesting and would have been clear if I was more interested in communicating and less interested in sounding clever.  Sorry =-(